### PR TITLE
imp: delegate store operation of `clientState` and `consensusState` in `initialize` to contract

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

This implements for now only the ibc-go part. Tests are expected to fail; we need new versions of contracts that handle setting the client state and consensus state.

closes: #3956 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by crodriguezvega** [comment]
> I've updated the contract according to the changes, but [these](https://github.com/cosmos/ibc-go/blob/ac18a2b075c2ea222406db92ec2dcf3ee9ddfa10/modules/light-clients/08-wasm/types/client_state_test.go#L236) [two](https://github.com/cosmos/ibc-go/blob/ac18a2b075c2ea222406db92ec2dcf3ee9ddfa10/modules/light-clients/08-wasm/types/client_state_test.go#L275) tests are failing because of an incorrect data, please check.
> 
> Can I push the new contracts to the repo so the CI can pass?

Thank you, @vmarkushin. I just gave you write permissions, so feel free to push to any of the `08-wasm` branches. When you upload the contracts, I will check those tests.

**Comment by vmarkushin** [comment]
> Thank you, @vmarkushin. I just gave you write permissions, so feel free to push to any of the `08-wasm` branches. When you upload the contracts, I will check those tests.

Deal, thanks!

**Review by DimitrisJim — APPROVED** [review]
one minor observation, would be nice to open separate issue about moving all these in a common file & documenting them.

File: modules/light-clients/08-wasm/types/client_state.go [hunk 1]
```diff
@@ -110,19 +110,32 @@ func (cs ClientState) GetTimestampAtHeight(
 	return consState.GetTimestamp(), nil
 }
 
+type instantiateMessage struct {
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/light-clients/08-wasm/types/client_state.go:113]: I believe this is superseded by the work in #4133

**Review by damiannolan — APPROVED** [review]

File: modules/light-clients/08-wasm/types/client_state.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/types/client_state.go:113]: Correct, all the messages are coming together in #4133

**Pull request merged**